templates/dhcpd.conf.erb

```diff
@@ -17,7 +17,7 @@ max-lease-time <%= @max_lease_time %>;
 authoritative;
 <% end -%>
 
-<% if has_variable?( 'dnsupdatekey' ) and @dnsupdatekey -%>
+<% if has_variable?( 'dnsupdatekey' ) and @dnsupdatekey or has_variable?( 'ddns_updates' ) and @ddns_updates -%>
```

Comment:
1. Use brackets here to contain the two parts of the expression, it's dependent on precedence of `and`/`or` right now.
2. Prefer `&&` and `||` in Ruby/ERB expressions, not `and`/`or`.